### PR TITLE
Enable nullability in EnumUnknown

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in EnumUnknown.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6407)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.EnumUnknown.cs [hunk 1]
```diff
@@ -13,21 +11,21 @@ public abstract partial class AxHost
...
+            private int _loc;
+            private readonly int _size;
 
             internal EnumUnknown(object[] arr)
             {
                 //if (AxHTraceSwitch.TraceVerbose) Debug.WriteObject(arr);
-                this.arr = arr;
-                loc = 0;
-                size = (arr is null) ? 0 : arr.Length;
+                _arr = arr;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.EnumUnknown.cs:21]: seems `arr `can be null

**Review by dreddy-work — CHANGES_REQUESTED** [review]

**Review by dreddy-work — APPROVED** [review]

**Comment by gpetrou** [comment]
@dreddy-work maybe we could merge this now that the build completed successfully?

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.EnumUnknown.cs [hunk 2]
```diff
@@ -48,13 +46,16 @@ unsafe HRESULT Ole32.IEnumUnknown.Next(uint celt, IntPtr rgelt, uint* pceltFetch
                 }
                 else
                 {
-                    for (; _location < _size && fetched < celt; ++_location)
+                    if (_array is not null)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.EnumUnknown.cs:49]: This is unnecessary check. If location < _size,   why would _array be null. Old code was not checking for it either.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/AxHost.EnumUnknown.cs]: I used `!` instead.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**